src/app/dataset-view/additional-components/dataset-settings-component/tabs/variables-and-secrets/components/edit-key-value-modal/edit-key-value-modal.component.html

```diff
@@ -47,12 +56,12 @@
             <label for="" class="form-label">Value <span class="text-danger">*</span></label>
             <div class="input-group">
                 <input
-                    [type]="row?.isSecret && !isShowExposedValue ? 'password' : 'text'"
+                    [type]="isSecretControl.value && !isShowExposedValue ? 'password' : 'text'"
                     class="form-control"
                     placeholder="Please enter value..."
                     formControlName="value"
                     data-test-id="value"
-                    [readonly]="row?.isSecret && !isShowExposedValue ? true : false"
+                    [readonly]="row && row.isSecret ? true : false"
```

Comment:
Not sure I understand this.

If `row` is not null this means we're in the "edit" mode, right?

Why are we making value read-only when editing secrets?

Reply:
My understanding was that secrets are extractable (can be viewed after creation) and editable. Correct me if I'm wrong @rmn-boiko

Reply:
Yes, I forgot that we can edit secret values. Then I delete the attribute **readonly**